packages/node_modules/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -386,17 +426,31 @@ export default class Meeting extends StatelessSparkPlugin {
   /**
    *
    * Leave the current meeting
+   * @param {Object} options An object of options
+   * @param {Boolean} options.keepLocalMedia Whether to tear down the local media connections or not.
    * @returns {Promise}
    * @memberof Meeting
    */
   leave() {
-    return this.meetingRequest.leaveMeeting({
-      locusUrl: this.locusUrl,
-      selfId: this.selfId,
-      correlationId: this.correlationId,
-      resourceId: null,
-      deviceUrl: this.deviceUrl
-    });
+    return this.meetingRequest
+      .leaveMeeting({
+        locusUrl: this.locusUrl,
+        selfId: this.selfId,
+        correlationId: this.correlationId,
+        resourceId: null,
+        deviceUrl: this.deviceUrl
+      })
+      .then(() => {
+        this.unsetLocalStream();
+        this.unsetRemoteStream();
+        this.unsetPeerConnections();
+      })
+      .then(() => {
+        this.roap.stop(this.correlationId, this.roapSeq);
```

Comment:
we should send an event when the roap is shut down for a meeting ?

Reply:
Sure. Though we need to document where we are sending all these events from so the end users will know what they have available. I'll add a section to the README

Reply:
What do you want the event name to be? I was going to move it into roap/handler.js

Reply:
open to ideas, i think we just need placeholders right now.